mlir/lib/Conversion/TosaToRock/TosaToRock.cpp

```diff
@@ -61,7 +61,7 @@ static bool isConstantZero(Value v) {
 }
 
 static Value expandTensor(ConversionPatternRewriter &rw, Operation *op,
-                          Value operand, SmallString<5> &layout,
+                          Value operand, SmallString<7> &layout,
```

Comment:
Bigger string for layouts with more dimensions.  I don't like this much, but I didn't find the correct ArrayRef or something else to generalise it.

Reply:
On further study, I can't make these SmallStrings into MutableArrayRef in expandTensor(), because part of the expansion is to insert a "g" into the layout string. As a MutableArrayRef, I can modify the existing elements, but since the buffer is owned by somebody else, I can't expand it. Therefore, SmallString<8> all around.